src/main/java/com/github/princesslana/eriscasper/action/GatewayAction.java

```diff
@@ -0,0 +1,23 @@
+package com.github.princesslana.eriscasper.action;
+
+import com.github.princesslana.eriscasper.gateway.OpCode;
+import io.reactivex.Completable;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class GatewayAction<I> implements Action {
+
+  public abstract OpCode getCode();
+
+  // There is no gateway action which takes no data
```

Comment:
I feel this comment is redundant. It's implied by not using Optional.

Reply:
Gotcha, I was just defining why it was different from `RouteAction`